axelrod/strategies/punisher.py

```diff
@@ -139,9 +139,9 @@ class LevelPunisher(Player):
     }
 
     def strategy(self, opponent):
-        if len(opponent.history) <= 10:
+        if len(opponent.cooperations) <= 10:
```

Comment:
So one of the errors you are getting is because `opponent.cooperations` is an integer, which show the number of times the opponent cooperated, thus it does not have `len()`. 

Here is a cheat sheet:
- `opponent.history` show all the moves made by the opponent until this point
- `len(opponent.history)` can show how many turns have been played 
- `opponent.cooperations` the number of times the opponent cooperated. 

What I am guessing your strategy does is:
```
If we played less than 10 turns -> defect
else if  the (num_turns_until_now - num_of_copperations)/num_turns_until_now > 0.2 -> cooperate
```

Hope this helps.